### PR TITLE
Added r-bsvars as a package

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thewchan** [comment]
Awaiting: https://github.com/conda-forge/staged-recipes/pull/26800, https://github.com/conda-forge/staged-recipes/pull/26801, and https://github.com/conda-forge/staged-recipes/pull/26802

File: recipes/r-bsvars/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+build:
+  merge_build_host: True  # [win]
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+  missing_dso_whitelist:
+    - '*/R.dll'        # [win]
+    - '*/Rblas.dll'    # [win]
+    - '*/Rlapack.dll'  #
```
Comment (mfansler) [inline recipes/r-bsvars/meta.yaml:24]: typo

```suggestion
    - '*/Rlapack.dll'  # [win]
```

File: recipes/r-bsvars/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    - cross-r-base {{ r_base }}    # [build_platform != target_platform]
+  host:
+    - r-base
+    - r-gigrvg
+    - r-r6
+    - r-rcpp >=1.0.7
+    - r-rcpparmadillo
+    - r-rcppprogress >=0.1
+    - r-rcpptn
+    - r-stochvol
```
Comment (mfansler) [inline recipes/r-bsvars/meta.yaml:48]: Needs blas and lapack

```suggestion
    - r-stochvol
    - libblas
    - liblapack
```

**Review by mfansler — CHANGES_REQUESTED** [review]

File: recipes/r-bsvars/meta.yaml (hunk 2 above)
Comment (thewchan) [reply recipes/r-bsvars/meta.yaml:48]: if BLAS is needed do I need to remove it from the missing DSO whitelist?

Comment (mfansler) [reply recipes/r-bsvars/meta.yaml:48]: No, it is still required. Windows build will still link to the Rblas/Rlapack, but that should itself dynamically link to whatever implementation the user chooses to install at runtime.

File: recipes/r-bsvars/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    - libblas
+    - liblapack
+  run:
+    - r-base
+    - {{ native }}gcc-libs         # [win]
```
Comment (mfansler) [inline recipes/r-bsvars/meta.yaml:53]: Last thing needed for win:

```suggestion
    - {{ native }}gcc-libs           # [win]
    - {{ native }}libwinpthread-git  # [win]

**Review by mfansler — CHANGES_REQUESTED** [review]

**Comment by thewchan** [comment]
Thanks @mfansler, anything else needed?

**Review by mfansler — APPROVED** [review]
Looks good - thanks!

**Pull request merged**